Filter auto

yerramshilpa opened the pull request:
<!--- Provide a brief but meaningful summary of your changes in the Title -->
<!-- The title will be used to automatically generate release notes through gren -->
<!-- 1. Include the JIRA ticket number in the title (e.g. IAE-500) -->
<!-- 2. Start the title with a verb (e.g. Change header styles) -->
<!-- 3. Use the imperative mood in the title (e.g. Fix, not Fixed or Fixes header styles) -->
<!-- 4. Use labels wisely and assign one label per issue. gren has the option to ignore issues that have one of the specified labels. -->

## Description
<!--- Describe your changes in detail -->

## Motivation and Context
<!-- If there is no existing JIRA ticket or Github issue, please provide why this change is required and what problem it solves -->
<!--- Otherwise, link to the ticket or issue here. -->

## Type of Change (Select One and Apply Github Label)
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue) -> Apply bugfix label
- [x] New feature (non-breaking change which adds functionality) -> Apply enhancement label
- [ ] Breaking change (fix or feature that would cause existing functionality to change) -> Apply breaking label

## Screenshots (if appropriate):

## Which browsers have you tested?
- [ ] Internet Explorer 11
- [ ] Edge
- [x] Chrome
- [ ] Firefox
- [x] Safari

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md] document.
- [x] My code passes the automated linter.
- [ ] This code has been reviewed by another team member and passes the reviewer checklist found in (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md]
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My code is 508 compliant as tested by AMP and JAWS
- [ ] Any dependent changes have been merged and published in downstream modules

davereed commented on libs/sam-formly-sample/src/lib/feature/formly-sample/services/autocomplete-sample.service.ts:
```diff
@@ -0,0 +1,68 @@
+/* tslint:disable */
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { SDSAutocomplteServiceInterface, SDSHiercarchicalServiceResult } from '@gsa-sam/components';
```
Spelling issue:
SDSAutocomplteServiceInterface

yerramshilpa replied:
This is the component service by Thomas. Can we discuss in the parling lot.

davereed commented on libs/sam-formly-sample/src/lib/feature/formly-sample/services/autocomplete-sample.service.ts:
```diff
@@ -0,0 +1,68 @@
+        if (maxSectionPosition > totalItemCount) {
+            maxSectionPosition = totalItemCount;
+        }
+        let subItemsitems = items.slice(currentItems, maxSectionPosition);
```
Use a more readable variable name for subItemsitems

sectionItems maybe?

yerramshilpa replied:
updated..

davereed commented on libs/sam-formly-sample/src/lib/feature/formly-sample/formly-input.component.ts:
```diff
@@ -1,144 +1,162 @@
+    this.settings3.debounceTime = 0;
+
+
+    this.autocompleteModel3.items.push(this.data[0]);
```
Accessing arrays like this is generally unsafe. Prefer to use lodash/underscore .head method.

    function head(array) {
      return (array && array.length) ? array[0] : undefined;
    }

davereed commented on libs/sam-formly-sample/src/lib/feature/formly-sample/formly-input.component.ts:
```diff
@@ -1,144 +1,162 @@
     {
-      key: 'testings',
+      key: 'filters',
       wrappers: ['accordianwrapper'],
```
Should be accordion not accordian

davereed commented:
Check / improve coverage on:
sds-filters-comunication.service.ts
autocomplete-sample.service.ts

thomastighe commented:
Also why are you updating the CSS when there is another story for that?

yerramshilpa commented:
> Also why are you updating the CSS when there is another story for that?

@thomastighe I thought you already know what I have updated it, but it seems you didn't get what I have updated it. I have not added any new CSS, I have updated with pre-existing styles which you should have done it earlier. There are other CSS works need to be done, those can be done on the ticket.

thomastighe commented:
> 
> 
> > Also why are you updating the CSS when there is another story for that?
> 
> @thomastighe I thought you already know what I have updated it, but it seems you didn't get what I have updated it. I have not added any new CSS, I have updated with pre-existing styles which you should have done it earlier. There are other CSS works need to be done, those can be done on the ticket.

@yerramshilpa  there is an entire story to do the styles IAE-21642 holistically, Which is also assigned to Carlos this sprint.

yerramshilpa commented on libs/packages/components/src/lib/selected-result/selected-result.component.spec.ts:
```diff
@@ -5,6 +5,10 @@ import { SDSSelectedItemModel } from './models/sds-selectedItem.model';
 import { SDSSelectedResultConfiguration } from './models/SDSSelectedResultConfiguration';
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+import { RouterModule } from '@angular/router';
```
removed

yerramshilpa commented on libs/packages/components/src/lib/selected-result/selected-result.component.html:
```diff
@@ -1,19 +1,21 @@
-    <div [attr.class]="disabled?'selected-item selected-item-disabled' :'selected-item'">
-      <ng-container *ngIf="itemTemplate" [ngTemplateOutlet]="itemTemplate"
-        [ngTemplateOutletContext]="{$implicit:result}">
+    <div>
```
removed due to style changes, no need to disable the div, instead we can disable span (CHIP and Button).

yerramshilpa commented on libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.ts:
```diff
@@ -1,4 +1,4 @@
-import { Component, Input, ViewChild, TemplateRef, ElementRef, forwardRef } from '@angular/core';
+import { Component, Input, ViewChild, TemplateRef, ElementRef, forwardRef, OnInit } from '@angular/core';
```
removed.

yerramshilpa commented on libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.html:
```diff
@@ -1,8 +1,8 @@
-      (keydown)="onKeydown($event)" [(ngModel)]="inputValue" autocomplete="off" role="combobox" aria-autocomplete="list"
-      aria-expanded="false" aria-haspopup="true" [attr.aria-owns]="configuration.id+ '-listbox'" [attr.placeholder]="configuration.autocompletePlaceHolderText">
+    <input [disabled]="disabled" (ngModelChange)="textChange($event)"  class="usa-input margin-bottom-2" #input [attr.id]="configuration.id" autocomplete="off" type="text" (focus)="inputFocusHandler()"
+      (keyup)="onKeydown($event)" [(ngModel)]="inputValue" autocomplete="off" role="combobox" aria-autocomplete="list"
```
before you made changes, I was working on the file it was a merge conflict issue.

thomastighe replied:
Can you please change this back to keydown

yerramshilpa commented on libs/packages/components/src/lib/public-api.ts:
```diff
@@ -4,7 +4,7 @@ export * from './toolbar/toolbar.module';
 export * from './footer/footer.module';
 export * from './pagination/pagination.module';
-
+export * from './autocomplete/autocomplete.component';
```
I was using the individual component in viewchild, if we don't export it was not working.

thomastighe replied:
@davereed  Am I missing something if you import the module into the sam formly module this should not need to be done. I could be mistaken on this but I do not think this should be done.

davereed replied:
Looks like she tried it the other way and it didn't work. Let's revisit this in the IP sprint to see if we can avoid importing the component and prefer to import the module only.

yerramshilpa commented on libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.spec.ts:
```diff
@@ -67,21 +67,18 @@ describe('SamAutocompleteComponent', () => {
+    
+    const event ='Level 7';
     component.textChange(event);
+    component.configuration.minimumCharacterCountSearch = 3;
```
reverted changes

thomastighe commented on libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.spec.ts:
```diff
@@ -67,21 +67,18 @@ describe('SamAutocompleteComponent', () => {
     fixture.detectChanges();
     const list = fixture.debugElement.query(By.css('.autocomplete-result'));
     expect(list.nativeElement.children.length).toBe(1);
-    const emptyItem = fixture.debugElement.query(By.css('.emptyResults'));
```
You did not address this being removed

yerramshilpa replied:
I already reverted those changes

yerramshilpa commented:
> > > Also why are you updating the CSS when there is another story for that?
> > 
> > 
> > @thomastighe I thought you already know what I have updated it, but it seems you didn't get what I have updated it. I have not added any new CSS, I have updated with pre-existing styles which you should have done it earlier. There are other CSS works need to be done, those can be done on the ticket.
> 
> @yerramshilpa there is an entire story to do the styles IAE-21642 holistically, Which is also assigned to Carlos this sprint.

@thomastighe Carlos updated the styles
